File: examples/pizzas_static_go/domain/placed_order.go
package domain

import "time"

type PlacedOrder struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	CustomerName string `json:"customer_name"`
	PizzaId string `json:"pizza_id"`
	Quantity int64 `json:"quantity"`
	Items []OrderItem `json:"items"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e PlacedOrder) EventName() string { return "PlacedOrder" }

func (e PlacedOrder) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/pizzas_static/lib/pizzas_domain/order/events/placed_order.rb
module PizzasDomain
  class Order
    module Events
      class PlacedOrder
        attr_reader :aggregate_id, :customer_name, :pizza_id, :quantity, :items, :status, :occurred_at

        def initialize(aggregate_id: nil, customer_name: nil, pizza_id: nil, quantity: nil, items: nil, status: nil)
          @aggregate_id = aggregate_id
          @customer_name = customer_name
          @pizza_id = pizza_id
          @quantity = quantity
          @items = items
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
